### ISSUE TITLE
Error starting REPL in server mode

### ISSUE THREAD

**Issue by GiggleLiu** [JuliaLang/julia#39070]
The `start_repl_server` function is not correctly implemented. The  listen function does not support function as the first argument.

https://github.com/JuliaLang/julia/blob/5ee2d601edcf2dcdd467041ba9f08c43bf23e1a3/stdlib/REPL/src/REPL.jl#L1235

## MWE
```julia
julia> using REPL

julia> REPL.start_repl_server(7777)
ERROR: MethodError: no method matching listen(::REPL.var"#80#81", ::Int64)
Closest candidates are:
  listen(::Sockets.IPAddr, ::Integer; backlog) at /home/leo/packages/julias/julia-1.5/share/julia/stdlib/v1.5/Sockets/src/Sockets.jl:619
  listen(::Any; backlog) at /home/leo/packages/julias/julia-1.5/share/julia/stdlib/v1.5/Sockets/src/Sockets.jl:612
Stacktrace:
 [1] start_repl_server(::Int64) at /home/leo/packages/julias/julia-1.5/share/julia/stdlib/v1.5/REPL/src/REPL.jl:1254
 [2] top-level scope at REPL[2]:1

julia> using Sockets

help?> Sockets.listen
  listen([addr, ]port::Integer; backlog::Integer=BACKLOG_DEFAULT) -> TCPServer

  Listen on port on the address specified by addr. By default this listens on
  localhost only. To listen on all interfaces pass IPv4(0) or IPv6(0) as
  appropriate. backlog determines how many connections can be pending (not
  having called accept) before the server will begin to reject them. The
  default value of backlog is 511.

  ────────────────────────────────────────────────────────────────────────────

  listen(path::AbstractString) -> PipeServer

  Create and listen on a named pipe / UNIX domain socket.
```

**Comment by mgkuhn** [comment]
`REPL.start_repl_server` was added in March 2014 to release-0.3, and apparently came from @keno's https://github.com/Keno/REPL.jl.

[Julia 0.3 base/socket.jl](https://github.com/JuliaLang/julia/blob/release-0.3/base/socket.jl) had an undocumented listen method with a call-back function as its first argument:
```
listen(cb::Callback,args...; backlog::Integer=BACKLOG_DEFAULT) = (sock=listen(args...;backlog=backlog);sock.ccb=cb;sock)
listen(cb::Callback,sock::Socket; backlog::Integer=BACKLOG_DEFAULT) = (sock.ccb=cb;listen(sock;backlog=backlog))
```
That may be what once was being called here.

In `REPL.start_repl_server`, you may have stumbled over a function that hasn't been used or tested for a long time.

The question is: should we fix or remove it? What was once the purpose of it?

Does the following do what you wanted?
```
using REPL
using Sockets

function start_repl_server(port::Int)
    server = listen(port)
    while true
        client = accept(server)
        REPL.run_repl(client)
    end
end

start_repl_server(7777)
```

**Comment by Keno** [comment]
Yeah, I don't think anybody has tried this in forever. Probably best handled in a package these days.

**Comment by mkitti** [comment]
Closed by #39162

**Comment by GiggleLiu** [comment]
Nice, I guess it is good to close this issue now. Thank @mgkuhn for your suggested implementation.
